ios/Classes/TiNfcNfcAdapterProxy.m

```diff
@@ -112,6 +170,28 @@ - (void)readerSession:(NFCNDEFReaderSession *)session didInvalidateWithError:(NS
       NO);
 }
 
+#pragma mark NFCReaderSessionDelegate
+
+- (void)tagReaderSession:(NFCTagReaderSession *)session didInvalidateWithError:(NSError *)error
+{
+  [self fireEvent:@"didInvalidateWithError" withObject:@{ @"error" : error.localizedDescription, @"cancelled" : @(error.code == 200) }];
+}
+
+- (void)tagReaderSessionDidBecomeActive:(NFCTagReaderSession *)session
+{
+  [self fireEvent:@"tagReaderSessionDidBecomeActive"];
+}
+
+- (void)tagReaderSession:(NFCTagReaderSession *)session didDetectTags:(NSArray<__kindof id<NFCTag>> *)tags
+{
+
+  id<NFCMiFareTag> tag = tags[0].asNFCMiFareTag;
+  [self fireEvent:@"didDetectTags"
```

Comment:
same here. check if we can use same approach which is already implemented before in ```- (void)readerSession:(NFCNDEFReaderSession *)session didDetectNDEFs:(NSArray<NFCNDEFMessage *> *)messages```